packages/block-directory/src/store/reducer.js

```diff
@@ -54,6 +61,30 @@ export const downloadableBlocks = ( state = {
 	return state;
 };
 
+/**
+ * Reducer returning an array of downloadable blocks.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export const errorNotices = ( state = {
+	notices: {},
+}, action ) => {
+	switch ( action.type ) {
+		case 'SET_ERROR_NOTICE_ID' :
+			return {
+				...state,
+				notices: Object.assign( {}, state.notices, {
+					[ action.blockId ]: action.noticeId,
+				} ),
```

Comment:
It's probably just my personal preference but I find `...` syntax easier to understand than `Object.assign` in these cases.

```js
return {
	...state,
	notices: {
		...state.notices,
		[ action.blockId ]: action.noticeId,
	}
};
```